src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -175,14 +184,36 @@ const SignOrExecuteForm = ({
     return id
   }
 
+  const onRelay = async () => {
+    const { gasLimit } = getTxOptions(advancedParams, currentChain)
+
+    const [, createdTx] = assertDependencies()
+
+    let txData
+    try {
+      txData = await prepareRelayTxData(createdTx, safe, proposeTx)
+    } catch (err) {
+      console.error('Error signing relay tx data', err)
+      return
+    }
+
+    try {
+      const relayResponse = await sponsoredCall({ chainId: safe.chainId, to: safeAddress, data: txData.data, gasLimit })
+      await dispatchTxRelay(relayResponse.taskId, txData.txId)
+    } catch (err) {
+      logError(Errors._631, (err as Error).message)
+      return
+    }
```

Comment:
As mentioned further up, let's keep the conditional signing here but move the rest to `dispatchTxRelay`.